exporters/trace/gcp_exporter/internal/recordable_test.cc

```diff
@@ -0,0 +1,114 @@
+#include "exporters/trace/gcp_exporter/recordable.h"
+#include <gtest/gtest.h>
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace gcp
+{
+
+TEST(Recordable, TestSetAttribute)
+{
+    Recordable rec;
+
+    // Set 'bool' type
+    const nostd::string_view bool_key = "bool_key";
+    const common::AttributeValue bool_value = true;
+    rec.SetAttribute(bool_key, std::move(bool_value));
+
+    // Set 'integer' type
+    const nostd::string_view int_key = "int_key";
+    const int64_t seven = 7;
+    const common::AttributeValue int_value = seven;
+    rec.SetAttribute(int_key, std::move(int_value));
+
+    // Set 'string' type
+    const nostd::string_view string_key = "string_key";
+    const common::AttributeValue string_value = "test";
+    rec.SetAttribute(string_key, std::move(string_value));
+
+    auto attr_map = rec.span().attributes().attribute_map();
+    
+    EXPECT_TRUE(attr_map["bool_key"].bool_value());
+    EXPECT_EQ(attr_map["int_key"].int_value(), seven);
+    EXPECT_EQ(attr_map["string_key"].string_value().value(), "test");
+}
+
+TEST(Recordable, TestSetIds)
+{
+    const opentelemetry::trace::TraceId trace_id(
+    std::array<const uint8_t, opentelemetry::trace::TraceId::kSize>(
+    {0, 1, 0, 3, 4, 8, 15, 34, 32, 78, 0, 0, 3, 2, 0, 1}));
+
+    const opentelemetry::trace::SpanId span_id(
+    std::array<const uint8_t, opentelemetry::trace::SpanId::kSize>(
+    {68, 23, 16, 0, 77, 2, 0, 2}));
+
+    const opentelemetry::trace::SpanId parent_span_id(
+    std::array<const uint8_t, opentelemetry::trace::SpanId::kSize>(
+    {4, 5, 0, 1, 1, 1, 1, 3}));
+
+    Recordable rec;
+
+    rec.SetIds(trace_id, span_id, parent_span_id);
+
+    EXPECT_EQ(rec.span().span_id(),
+            std::string(reinterpret_cast<const char *>(span_id.Id().data()), trace::SpanId::kSize));
+    EXPECT_EQ(rec.span().parent_span_id(),
+            std::string(reinterpret_cast<const char *>(parent_span_id.Id().data()), trace::SpanId::kSize));
+}
+
+
+TEST(Recordable, TestSetName)
+{
+    Recordable rec;
+    const nostd::string_view expected_name = "Test Span";
+    rec.SetName(expected_name);
+    EXPECT_EQ(rec.span().display_name().value(), expected_name);
```

Comment:
In the OTEL C++ project we usually always have the expected value as first argument. This is not part of the OTEL C++ repo, so you're free to adopt any standard you want. However, it would make your test code more readable to be consistent (sometimes you have the expected value first, sometimes the actual value).

Reply:
Sure!